extensions-support/azure-core-http-client-vertx/runtime/src/main/java/org/apache/camel/quarkus/support/azure/core/http/vertx/BufferedVertxHttpResponse.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.quarkus.support.azure.core.http.vertx;
+
+import java.io.ByteArrayInputStream;
+import java.io.InputStream;
+import java.nio.ByteBuffer;
+
+import com.azure.core.http.HttpRequest;
+import com.azure.core.http.HttpResponse;
+import io.vertx.core.buffer.Buffer;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+public final class BufferedVertxHttpResponse extends VertxHttpAsyncResponse {
+
+    private final Buffer body;
+
+    BufferedVertxHttpResponse(HttpRequest request, io.vertx.ext.web.client.HttpResponse response, Buffer body) {
+        super(request, response);
+        this.body = body;
```

Comment:
We can do something better here (not sure it's needed).

We could just get the body handler or the read stream associated with it and returns truly async Mono and Flux. It would only be useful if:

1. the operation needs to be lazy
2. the body size is huge

Reply:
Originally, I was thinking along the same lines. When I started testing with massive responses (10s of gigabytes) it seems Azure / Azure SDK handles it as `HTTP 206 Partial Content`, so the overall body is read and assembled in chunks. Thus I left things as-is.

> We could just get the body handler or the read stream associated with it

Do you have an example of that?

Reply:
@tsegismont - do you have an example using `pipe`? If not I can try to implement one.

Reply:
I would not recommend to use pipe with a custom implementation of `WriteStream`. If you face the need to do this, it means that you should probably be using the raw `HttpClient` which is more flexible than the `WebClient` in this area.

One thing you need to know about the `pipe` body codec: it is only when the whole content has been read that the `WebClient.send` callback is invoked. So you can't, for example, inspect the HTTP status code or headers before deciding what to do with the content.

I you choose to go the `HttpClient` route, I think there are examples in the Vert.x docs for consumption of HTTP bodies.

Reply:
Thanks @tsegismont!

So I believe we are good with the current solution for now. If later we need real streaming we would switch to the http client.